File: day2/wrapping.rb
packages = File.readlines('./input', :chomp=> true)
area = 0
length = 0
packages.each do |dimensions|
  dimensions = dimensions.split('x')
  dimensions.each_index { |i| dimensions[i] = dimensions[i].to_i }
  length += dimensions[0] * dimensions[1] * dimensions[2]
  area += 2*dimensions[0]*dimensions[1] + 2*dimensions[0]*dimensions[2] + 2*dimensions[1]*dimensions[2]
  dimensions.sort!.pop
  length += 2*(dimensions[0] + dimensions[1])
  area += dimensions[0] * dimensions[1]
end
puts "#{area} sqft of papaer"
puts "#{length} feet of ribbon"

File: day2/wrapping.py
if __name__ == "__main__":

    with open('input') as f:
        prezzies = f.readlines()

    def format_present(present):
        dims = present.split('x')
        dims = list(map(int,dims))
        dims.sort()
        return dims
    
    prezzies = list(map(format_present, prezzies))

    paper = 0
    ribbon = 0
    for p in prezzies:
        (l,w,h) = p
        paper = paper + 2*l*w + 2*l*h + 2*w*h + l*w
        ribbon = ribbon + 2*(l+w) + l*w*h
    print(paper)
    print(ribbon)